core-blocks/playlist/index.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	Toolbar,
+} from '@wordpress/components';
+import { pick } from 'lodash';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	editorMediaUpload,
+	MediaUpload,
+	RichText,
+	BlockControls,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import './editor.scss';
+
+export const name = 'core/playlist';
+
+export const settings = {
+	title: __( 'Playlist' ),
+
+	description: __( 'The Playlist block allows you to embed playlist files and play them back using a Core playlist.' ),
+
+	icon: 'format-audio',
+
+	category: 'common',
+
+	attributes: {
+		ids: {
+			type: 'array',
+		},
+		src: {
+			type: 'string',
+		},
+		type: {
+			type: 'string',
+		}
+	},
+
+	supports: {
+		align: true,
+	},
+
+
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+			// edit component has its own src in the state so it can be edited
+			// without setting the actual value outside of the edit UI
+			this.state = {
+				editing: ! this.props.attributes.src,
+				src: this.props.attributes.src,
+			};
+		}
+
+		render() {
+			// const { registerBlockType } = wp.blocks;
```

Comment:
Is this leftover from testing?

Reply:
I cleaned this up a bunch. Thanks! My PR is updated. Continuing to plug away at the edit state issue with the ServerSideRender component